File: backend/src/OrderTracking.Domain/Entities/Order.cs
using OrderTracking.Domain.Enums;

namespace OrderTracking.Domain.Entities;

/// <summary>
/// Представляет заказ в системе отслеживания заказов.
/// </summary>
public class Order
{
    /// <summary>
    /// Уникальный идентификатор заказа.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Номер заказа.
    /// </summary>
    public string OrderNumber { get; set; } = string.Empty;

    /// <summary>
    /// Описание заказа.
    /// </summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// Текущий статус заказа.
    /// </summary>
    public OrderStatus Status { get; set; }

    /// <summary>
    /// Дата и время создания заказа.
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// Дата и время последнего обновления заказа.
    /// </summary>
    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/order.ts
export enum OrderStatus {
  Created = 0,
  Sent = 1,
  Delivered = 2,
  Cancelled = 3,
}

export interface OrderDto {
  id: string;
  orderNumber: string;
  description: string;
  status: OrderStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderDto {
  orderNumber: string;
  description: string;
}

export interface UpdateOrderStatusDto {
  status: OrderStatus;
}

export interface OrderStatusChangedEvent {
  orderId: string;
  orderNumber: string;
  previousStatus: OrderStatus;
  newStatus: OrderStatus;
  changedAt: string;
}
